src/credentials/AzureCredentialsConfig.ts

```diff
@@ -20,15 +20,16 @@ export function isCredentialsComplete(credentials: AzureCredentials, ignoreSecre
     case 'currentuser':
       return true;
     case 'clientsecret':
+    case 'clientsecret-obo':
       return !!(
         credentials.azureCloud &&
         credentials.tenantId &&
         credentials.clientId &&
         // When ignoreSecret is set we consider the credentials complete without checking the secret
         !!(ignoreSecret || credentials.clientSecret)
       );
-    default:
```

Comment:
I think we should keep the default behaviour to throw a meaningful error if we encounter an unexpected credential type.

Reply:
Thanks @aangelisc. Previously it was removed due to errors about the default being unreachable. It's added back now after extracting the authType into a separate variable.